lib/FrontendTool/FrontendTool.cpp

```diff
@@ -184,20 +185,39 @@ static bool emitMakeDependenciesIfNeeded(DiagnosticEngine &diags,
 }
 
 namespace {
+struct SwiftModuleTraceInfo {
+  std::string ModulePath;
+  bool IsImportedDirectly;
+  bool IsCompiledWithLibraryEvolution;
+};
+
 struct LoadedModuleTraceFormat {
   std::string Name;
   std::string Arch;
-  std::vector<std::string> SwiftModules;
+  std::vector<SwiftModuleTraceInfo> SwiftModules;
 };
 }
 
 namespace swift {
 namespace json {
+template <> struct ObjectTraits<SwiftModuleTraceInfo> {
+  static void mapping(Output &out, SwiftModuleTraceInfo &contents) {
+    out.mapRequired("path", contents.ModulePath);
+    out.mapRequired("is-imported-directly", contents.IsImportedDirectly);
```

Comment:
Let's go with snake case or camel case instead of kebab case. That's probably slightly easier for other tools to handle.